src/backend/joanie/tests/testing_utils.py

```diff
@@ -736,151 +738,173 @@ def generate_simple(self):  # pylint: disable=too-many-locals,too-many-statement
+
+        if "Certificate product with discount" in items:
+            course_run_certificate_discount = factories.CourseRunFactory(
+                title="Course run certificate discount",
+                resource_link=OPENEDX_COURSE_RUN_URI.format(
+                    course="00001", course_run="CourseRunCertificate_run1"
+                ),
+                course__code="00001",
+                course__title="Course for certificate discount",
+                # Give access to organization owner user
+                course__users=[[organization_owner, enums.OWNER]],
+                course__organizations=[organization],
+                languages=self.get_random_languages(),
+                state=CourseState.ONGOING_OPEN,
+                is_listed=True,
+            )
+            certificate_product_discount = factories.ProductFactory(
+                type=enums.PRODUCT_TYPE_CERTIFICATE,
+                title="Certificate Product with discount",
+                courses=[course_run_certificate_discount.course],
+                target_courses=[],
+                certificate_definition=factories.CertificateDefinitionFactory(
+                    title="Certification discount",
+                    name="Become a discounted learner certificate",
+                ),
+                price=100,
+            )
+            certificate_discount_offering = models.CourseProductRelation.objects.get(
+                course=certificate_product_discount.courses.first(),
+                product=certificate_product_discount,
+            )
+            factories.OfferingRuleFactory(
+                course_product_relation=certificate_discount_offering,
+                discount=discount,
+            )
+            self.log(
+                f'Successfully created "{certificate_product_discount.title}" product'
+            )
+            course_run_certificate_discount.save()
 
-        course_run_certificate.save()
-        course_run_certificate_discount.save()
-        course_run_credential.save()
-        course_run_credential_discount.save()
+        if "Credential product" in items:
+            course = factories.CourseFactory(
+                code="00002",
+                title="Course for credential",
+                # Give access to organization owner user
+                users=[[organization_owner, enums.OWNER]],
+                organizations=[organization],
+            )
+            self.log(f'Successfully created "{course.title}" course')
+            self.log("")
+
+            course_run_credential = factories.CourseRunFactory(
+                title="Course run credential",
+                resource_link=OPENEDX_COURSE_RUN_URI.format(
+                    course="00002", course_run="CourseRunCredential_run1"
+                ),
+                course=course,
+                languages=self.get_random_languages(),
+                state=CourseState.ONGOING_OPEN,
+                is_listed=False,
+            )
```

Comment:
```suggestion
            course_run_credential = factories.CourseRunFactory(
                title="Course run credential",
                resource_link=OPENEDX_COURSE_RUN_URI.format(
                    course="00002", course_run="CourseRunCredential_run1"
                ),
                course__code="00002",
                course__title="Course for credential",
                course__users=[[organization_owner, enums.OWNER]],
                course__organizations=[organization],
                languages=self.get_random_languages(),
                state=CourseState.ONGOING_OPEN,
                is_listed=False,
            )
```
Just to match the way you prepared the other course runs

Reply:
In fact, we should create the course first, just to match the reality.
I changed the others.